pkg/capr/planner/certificaterotation.go

```diff
@@ -139,20 +142,81 @@ func (p *Planner) rotateCertificatesPlan(controlPlane *rkev1.RKEControlPlane, to
 		Content: base64.StdEncoding.EncodeToString([]byte(idempotentRotateScript)),
 		Path:    rotateScriptPath,
 	})
-	rotatePlan.Instructions = append(rotatePlan.Instructions, []plan.OneTimeInstruction{
-		{
-			Name:    "rotate certificates",
-			Command: "sh",
-			Args:    args,
+	rotatePlan.Instructions = append(rotatePlan.Instructions, plan.OneTimeInstruction{
+		Name:    "rotate certificates",
+		Command: "sh",
+		Args:    args,
+	})
+	if isControlPlane(entry) {
+		if kcmCertDir := getArgValue(config[KubeControllerManagerArg], CertDirArgument, "="); kcmCertDir != "" && getArgValue(config[KubeControllerManagerArg], TLSCertFileArgument, "=") == "" {
+			rotatePlan.Instructions = append(rotatePlan.Instructions, []plan.OneTimeInstruction{
+				{
+					Name:    "remove kube-controller-manager cert for regeneration",
+					Command: "rm",
+					Args: []string{
+						"-f",
+						fmt.Sprintf("%s/%s", kcmCertDir, DefaultKubeControllerManagerCert),
+					},
+				},
+				{
+					Name:    "remove kube-controller-manager key for regeneration",
+					Command: "rm",
+					Args: []string{
+						"-f",
+						fmt.Sprintf("%s/%s", kcmCertDir, strings.ReplaceAll(DefaultKubeControllerManagerCert, ".crt", ".key")),
+					},
+				},
+			}...)
+			if runtime == capr.RuntimeRKE2 {
+				rotatePlan.Instructions = append(rotatePlan.Instructions, plan.OneTimeInstruction{
+					Name:    "remove kube-controller-manager static pod manifest",
+					Command: "rm",
+					Args: []string{
+						"-f",
+						"/var/lib/rancher/rke2/agent/pod-manifests/kube-controller-manager.yaml",
+					},
+				})
```

Comment:
Does RKE2 add this file back automatically? Should we delete the running container instead of the manifest file?

Reply:
RKE2 adds this file on startup, which will happen during certificate rotation.

This is the contractual method of managing the static pod component as per the RKE2 team.